programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -367,25 +379,21 @@ impl<'a, 'info> ScanningAccountRetriever<'a, 'info> {
         let serum3_start = perp_oracles_start + n_perps;
         let n_serum3 = ais[serum3_start..]
             .iter()
-            .filter(|x| serum3_cpi::load_open_orders_ref(x).is_ok())
+            .filter(|x| x.data_len() == std::mem::size_of::<serum_dex::state::OpenOrders>() + 12)
```

Comment:
- `filter()` isn't the right choice here, maybe `.position(!is_serum_oo)`? 
- might have issues if an oracle randomly has the same size as a serum oo. serum oos always start with the "serum" string, let's check that too